python/ray/autoscaler/aws/config.py

```diff
@@ -250,7 +265,14 @@ def _configure_security_group(config):
             "IpRanges": [{
                 "CidrIp": "0.0.0.0/0"
             }]
-        }])
+        }]
+        try:
+            IpPermissions.extend(config["provider"]["extra_config"][
+                "security_group"]["IpPermissions"])
```

Comment:
While reusing the existing Boto3 IpPermission model here is quite powerful, having users write it directly into their config unfortunately forces them to pick through the (pretty gnarly) Boto3 IpPermissions docs and cross-reference them with other AWS docs to understand the significance of things like EC2-VPC vs. EC2-Classic parameters. Given the comparatively simple modern EC2-VPC IpPermissions model, the difficulty of trying to support/test Ray cluster configuration via deprecated technologies like EC2-Classic, and the difficulty involved in backing out of this decision in the future, I would recommend exposing our own simplified EC2-VPC abstraction of this model to users. For example: 
https://github.com/ray-project/ray/blob/207a0aa2fe477edb261de05cefaf7b3b4eed4bc5/python/ray/autoscaler/ray-schema.json#L15-L78 https://github.com/ray-project/ray/blob/207a0aa2fe477edb261de05cefaf7b3b4eed4bc5/python/ray/autoscaler/aws/example-security-groups.yaml#L18-L75.

Reply:
I agree this is a better approach than using the existing Boto3 IpPermission model.  Since you have written much of what's needed to support this, do you want to make a separate PR for this?

Reply:
Also do you have any purposed solutions to the issue you mentioned about the failure when unequal but equivalent rule is provided by the user? Maybe bringing the server-side coercion logic here could be a solution. Not sure how feasible that is though.

Reply:
Definitely! I actually opened up a PR for this yesterday: https://github.com/ray-project/ray/pull/7873. It would be great to get some reviewer eyes on it, since the high degree of manual effort currently involved in configuring AWS account Subnets and/or Security Group rules without these changes effectively presents a blocker to us standing up Ray Clusters at scale across separate internal AWS accounts.

Reply:
My changes currently only do naive IpPermission equality checks, and thus suffer from the same coercion corner case problems. However, I've been taking some time today to look through the server-side rule target coercion logic to see how easy it is to replicate within the AWS autoscaler code (and how fragile the resulting solution may be). If I can land on a promising solution, then I'll let you know and update the PR as well.